macOS/DuckDuckGo/Tab/View/BrowserTabViewController.swift

```diff
@@ -192,6 +192,7 @@ final class BrowserTabViewController: NSViewController {
     @objc
     private func windowWillClose(_ notification: NSNotification) {
         self.removeWebViewFromHierarchy()
+        self.newTabPageWebViewModel.removeUserScripts()
```

Comment:
I don't like this approach overall, the view model shouldn't necessarily have to care about breaking the retain cycle. However the issue is laid out in [this post](https://stackoverflow.com/a/26383032), where the content controller strongly retains its message handler. We have applied the same fix described in that post elsewhere in the codebase (look at `WeakScriptMessageHandlerBox`) but applying that here wasn't straightforward.

Overall this works and isn't actively harmful, but it's a bit of a bandaid rather than a fix at a lower level. Open to suggestions or strong feelings here. :)